src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -1194,24 +1199,9 @@ private void stripHopByHopHeaders(HttpHeaders headers) {
      * Miscellaneous
      **************************************************************************/
 
-    /**
-     * Tells the client that something went wrong trying to proxy its request. If the Bad Gateway is a response to
-     * an HTTP HEAD request, the response will contain no body, but the Content-Length header will be set to the
-     * value it would have been if this 502 Bad Gateway were in response to a GET.
-     *
-     * @param httpRequest the HttpRequest that is resulting in the Bad Gateway response
-     * @return true if the connection will be kept open, or false if it will be disconnected
-     */
-    private boolean writeBadGateway(HttpRequest httpRequest) {
```

Comment:
@viacheslav-fomin-main i see that the `respondWithShortCircuitResponse` method calls [`currentFilters.proxyToClientResponse`](https://github.com/viacheslav-fomin-main/LittleProxy/blob/bef40d77fd278a97c6d16c826937d63d4264cb73/src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java#L1262).

i'm wondering if it's possible for us to implement similar functionality in that method without making this change.

e.g.

```
@Override
public HttpObject handleProxyToClientResponse(HttpObject httpObject) {
  if (httpObect.statusCode() == HttpResponseStatus.BAD_GATEWAY) {
    // logic here
  }
}
```